sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/CheckAnalysis.scala

```diff
@@ -117,66 +117,72 @@ trait CheckAnalysis extends PredicateHelper {
                 failAnalysis(s"Window specification $s is not valid because $m")
               case None => w
             }
-          case s @ ScalarSubquery(query, conditions, _)
+
+          case e @ PredicateSubquery(query, _, _, _) =>
```

Comment:
It might be better to add a catch all `SubqueryExpression` case after the ScalarSubquery case, instead of adding one specifically aimed at a predicate subquery.